infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerInternalRuntime.java

```diff
@@ -186,15 +188,18 @@ protected void internalStart(Map<String, String> startOptions) throws Infrastruc
             String machineName = startQueue.peek();
             while (machineName != null) {
                 DockerContainerConfig container = dockerEnvironment.getContainers().get(machineName);
+                checkInterruption();
                 sendStartingEvent(machineName);
                 try {
                     if (restore) {
                         restoreMachine(machineName, container);
                     } else {
                         startMachine(machineName, container);
                     }
+                    checkInterruption();
                     sendRunningEvent(machineName);
                 } catch (InfrastructureException e) {
```

Comment:
Can you elaborate on what behavior would be if RuntimeStartInterruptedException is caught here